tests/validation/tests/v3_api/test_rke_cluster_provisioning.py

```diff
@@ -995,3 +999,94 @@ def create_and_validate_custom_host(node_roles, random_cluster_name=False):
         i += 1
     cluster = validate_cluster(client, cluster, k8s_version=K8S_VERSION)
     return cluster, aws_nodes
+
+
+def test_kdm_changes():
```

Comment:
Can you add a function description since this is such a complex function?

Reply:
Added